File: lib/src/main/java/com/sd/lib/http/exception/HttpExceptionResponseCode.java
package com.sd.lib.http.exception;

/**
 * Http返回码异常
 */
public class HttpExceptionResponseCode extends HttpException
{
    private final int mCode;
    private final String mDetails;

    public HttpExceptionResponseCode(int code, String details)
    {
        mCode = code;
        mDetails = details;
    }

    public static HttpExceptionResponseCode from(int code)
    {
        if (code >= 400)
        {
            return new HttpExceptionResponseCode(code, null);
        }
        return null;
    }

    public int getCode()
    {
        return mCode;
    }

    public String getDetails()
    {
        return mDetails;
    }
}

File: lib/src/main/java/com/sd/lib/http/exception/HttpExceptionResponseCode.kt
package com.sd.lib.http.exception

import android.content.Context
import com.sd.lib.http.R

/**
 * http返回码异常
 */
class HttpExceptionResponseCode : HttpException {
    val code: Int

    @JvmOverloads
    constructor(code: Int, message: String? = "") : super(message = message) {
        this.code = code
    }

    override fun getDescFormat(context: Context): String {
        return context.getString(R.string.lib_http_desc_exception_response_code, code)
    }

    override fun toString(): String {
        return "${super.toString()} code:${code}"
    }

    companion object {
        @JvmStatic
        fun from(code: Int): HttpExceptionResponseCode? {
            return if (code >= 400) {
                HttpExceptionResponseCode(code)
            } else {
                null
            }
        }
    }
}